homeassistant/components/reolink/host.py

```diff
@@ -327,6 +327,12 @@ def unregister_webhook(self):
 
     async def handle_webhook(
         self, hass: HomeAssistant, webhook_id: str, request: Request
+    ):
+        """Shield the incoming webhook callback from cancellation."""
+        await asyncio.shield(self.handle_webhook_shielded(hass, webhook_id, request))
```

Comment:
Why shield it and not just create it as a task and call it a day?
```suggestion
        self._hass.async_create_task(self.handle_webhook_shielded(hass, webhook_id, request))
```

Reply:
Because I got scared of this warning https://docs.python.org/3/library/asyncio-task.html:

> Important
> Save a reference to the result of this function, to avoid a task disappearing mid-execution. The event loop only keeps weak references to tasks. A task that isn’t referenced elsewhere may get garbage collected at any time, even before it’s done. For reliable “fire-and-forget” background tasks, gather them in a collection:

But I am guessing the hass version of async_create_task takes care of that?

Reply:
I am going to revert this suggestion:
In testing with `self.handle_webhook_shielded` I am getting tons of these messages at shutdown of homeassistant:
`2023-03-17 13:01:13.827 WARNING (MainThread) [homeassistant.core] Task <Task pending name='Task-1377' coro=<ReolinkHost.handle_webhook_shielded() running at /home/hass/home-assistant/homeassistant/components/reolink/host.py:345> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[set.remove()]> was still running after stage 2 shutdown; Integrations should cancel non-critical tasks when receiving the stop event to prevent delaying shutdown`

Furthermore the actual `handle_webhook_shielded` seems to start but then stop somewhere halfway because it is cancelled/supsended/garbage collected or something like that when using `self._hass.async_create_task`.

It was working perfectly fine withouth warnings when using `await asyncio.shield`